common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java

```diff
@@ -160,4 +166,146 @@ void testGetLong() throws Exception {
+            + "    \"type\": \"literal\",\n"
+            + "    \"dataType\": \"null\",\n"
+            + "    \"value\": \"null\"\n"
+            + "  },\n"
+            + "  \"lower\": {\n"
+            + "    \"type\": \"literal\",\n"
+            + "    \"dataType\": \"integer\",\n"
+            + "    \"value\": \"6\"\n"
+            + "  }\n"
+            + "}";
+    Assertions.assertEquals(
+        objectMapper.readValue(expected, RangePartitionDTO.class),
+        objectMapper.readValue(jsonValue, RangePartitionDTO.class));
+
+    partition =
+        ListPartitionDTO.builder()
+            .withName("p202204_California")
+            .withLists(
+                new LiteralDTO[][] {
+                  {
+                    new LiteralDTO.Builder()
+                        .withDataType(Types.DateType.get())
+                        .withValue("2022-04-01")
+                        .build(),
+                    new LiteralDTO.Builder()
+                        .withDataType(Types.StringType.get())
+                        .withValue("Los Angeles")
+                        .build()
+                  },
+                  {
+                    new LiteralDTO.Builder()
+                        .withDataType(Types.DateType.get())
+                        .withValue("2022-04-01")
+                        .build(),
+                    new LiteralDTO.Builder()
+                        .withDataType(Types.StringType.get())
+                        .withValue("San Francisco")
+                        .build()
+                  }
+                })
+            .build();
+    jsonValue = JsonUtils.objectMapper().writeValueAsString(partition);
+    expected =
+        "{\n"
+            + "  \"type\": \"list\",\n"
+            + "  \"name\": \"p202204_California\",\n"
+            + "  \"lists\": [\n"
+            + "    [\n"
+            + "      {\n"
+            + "        \"type\": \"literal\",\n"
+            + "        \"dataType\": \"date\",\n"
+            + "        \"value\": \"2022-04-01\"\n"
+            + "      },\n"
+            + "      {\n"
+            + "        \"type\": \"literal\",\n"
+            + "        \"dataType\": \"string\",\n"
+            + "        \"value\": \"Los Angeles\"\n"
+            + "      }\n"
+            + "    ],\n"
+            + "    [\n"
+            + "      {\n"
+            + "        \"type\": \"literal\",\n"
+            + "        \"dataType\": \"date\",\n"
+            + "        \"value\": \"2022-04-01\"\n"
+            + "      },\n"
+            + "      {\n"
+            + "        \"type\": \"literal\",\n"
+            + "        \"dataType\": \"string\",\n"
+            + "        \"value\": \"San Francisco\"\n"
+            + "      }\n"
+            + "    ]\n"
+            + "  ]\n"
+            + "}";
+    Assertions.assertEquals(
+        objectMapper.readValue(expected, ListPartitionDTO.class),
+        objectMapper.readValue(jsonValue, ListPartitionDTO.class));
+  }
```

Comment:
I think it would be better to add some negative cases where certain fields do not exist.

Reply:
added